[gh-pages] [ZEPPELIN-2597] updating navbar and main section

soralee opened the pull request:
## What is this PR for?
For more comfortable guidance, improved navigation bar, main blue section and what's new section.
And if anyone suggests idea or give me feedback, it would be appreciated!

1. In main blue section
 - add zeppelin logo
 - change buttons, `Get started` and `Download`
 - change button color.
 - change font style of description 
 - change title font-size more smaller
 - change background color for dividing navbar and this section.

2. regarding to improve navbar
 - improve: change position of items
 - improve: change font option in dropdown-menu
 - fix : when not active, revert color 
 - fix : when resizing window, dropdown-menu toggle text is disappeared.
 - fix : after using navigation bar in small window, and then drop-toggle place is misaligned in big window.

3. what's new sections
 - (minor) improve: aligned title text when small size windows


## What type of PR is it?
[ Bug Fix | Improvement | Documentation ]

## Todos
* describe above `What is this PR for?` part

## What is the Jira issue?
* [ZEPPELIN-2597](https://issues.apache.org/jira/browse/ZEPPELIN-2597)

## How should this be tested?
1. run `bundle exec jekyll serve --watch` in this branch
2. `open localhost:4000`

## Screenshots (if appropriate)
### 1. blue section
[Before]
![image](https://user-images.githubusercontent.com/8110458/26919645-640fe532-4c71-11e7-9073-8acfd6100d08.png)

[After]
![image](https://user-images.githubusercontent.com/8110458/26919850-023b7f96-4c72-11e7-9069-5d3e5cf77c46.png)

### 2. navbar
-  change position of items (Before -> After)

![image](https://user-images.githubusercontent.com/8110458/26920156-fdbff20c-4c72-11e7-8171-0e426889c3ef.png)
![image](https://user-images.githubusercontent.com/8110458/26920075-c23689da-4c72-11e7-95a5-7ab4e3fe4f5a.png)

---

- change font option in dropdown-menu (Before -> After)

![image](https://user-images.githubusercontent.com/8110458/26920330-a8caf336-4c73-11e7-8539-fd7604b7146f.png)
![image](https://user-images.githubusercontent.com/8110458/26920306-92a3fb66-4c73-11e7-965e-e24a9042f912.png)

---
 - fix : when not active, revert color 

![before](https://user-images.githubusercontent.com/8110458/26920990-a3476eb0-4c75-11e7-8ea6-c2d457f5dae7.gif)
![after](https://user-images.githubusercontent.com/8110458/26920991-a3581bfc-4c75-11e7-9640-d4aaa3a523cb.gif)

---

- fix : when resizing window, dropdown-menu toggle text is disappeared.

![image](https://user-images.githubusercontent.com/8110458/26920443-0541f2fe-4c74-11e7-8d69-b845cff872b8.png)
![image](https://user-images.githubusercontent.com/8110458/26920544-4c5120a2-4c74-11e7-91b1-e41f074b9b82.png)

---

 - fix : after using navigation bar in small window, and then drop-toggle place is misaligned in big window.

![before](https://user-images.githubusercontent.com/8110458/26920746-ecf31092-4c74-11e7-8d34-9ff01e6692a7.gif)
![after](https://user-images.githubusercontent.com/8110458/26920745-ecef8e18-4c74-11e7-967b-e05789f14d28.gif)

---

### 3. what's new sections
 - (minor) improve: aligned title text when small size windows

![image](https://user-images.githubusercontent.com/8110458/26921198-43f4c326-4c76-11e7-845d-07225cc64bf6.png)
![image](https://user-images.githubusercontent.com/8110458/26921228-5179a804-4c76-11e7-89a1-5251675090d1.png)



### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes, this is.

soralee commented:
Hi, @AhyoungRyu @1ambda @Leemoonsoo. Please review again :slightly_smiling_face:

soralee commented:
Some adding fix list
 - fix : when navbar-toggle in collapsed, do not revert color.
[Before]
![before](https://user-images.githubusercontent.com/8110458/26922864-63697cba-4c7b-11e7-96fb-824aaaf0b6d6.gif)
[After]
![after](https://user-images.githubusercontent.com/8110458/26922863-63383ea2-4c7b-11e7-8195-a7826a28cb6a.gif)

---
 - fix: not fully hover in dropdown-menu title

![before](https://user-images.githubusercontent.com/8110458/26923275-986a311a-4c7c-11e7-9043-ea559ed682ef.gif)
![after](https://user-images.githubusercontent.com/8110458/26923162-35bad1a0-4c7c-11e7-9bac-14ae118c229d.gif)

1ambda commented:
Let me check. Thanks for the contribution and great effort on the improving official website.

Leemoonsoo commented:
Cool!.

I tried this branch and found

![image](https://user-images.githubusercontent.com/1540981/26958347-ed2189cc-4c7e-11e7-8640-cd6c312caac4.png)

"Get Started" and "Download" button in the dark.
I think dark (or gray) button usually used to describe 'inactive' feature. So it may give wrong impression user. Can we choose other color that tells more 'active'?

1ambda commented:
Then bright white color would be better for the buttons.

1ambda commented on _includes/themes/zeppelin/_navigation.html:
```diff
@@ -1,81 +1,98 @@
+    <div style="margin-top: -250px; position: relative;">
+      <h1 style="font-size:40px;">Apache Zeppelin</h1>
+      <p class="description">
+        A web-based notebook that enables interactive data analytics for
```
1. Text for Section 1

A. Remove **interactive** since it's duplicated with `interactive and ...`
B. Or use the original text

> A web-based notebook that enables interactive data analytics. 
You can make beautiful data-driven, interactive and collaborative documents with SQL, Scala and more.

But +1 for **A** because it's simpler and short. IMO, we don't need some words like `you` and `make` in the front page title.

soralee replied:
I removed `interactive and ...`. now, you can confirm this.

1ambda replied:
i couldn't see the commit for the change. Could you link that one?

soralee replied:
Sure, https://github.com/apache/zeppelin/pull/2401/commits/110ead1734d5ec2d48d0b38faefb2dd47b1254a0

1ambda commented:
2. (BUG) Navbar doesn't fold when I click other a component.

![2401_nav_folded_always](https://user-images.githubusercontent.com/4968473/26961060-f5a002fc-4d16-11e7-99e0-907d32686e68.gif)

1ambda commented:
3. Can't find `favicon.ico`.

When I execute `bundle exec jekyll serve --watch`, I get this error.

```
[2017-06-09 13:13:01] ERROR `/favicon.ico' not found.
```

Do we have the favicon for Zeppelin?

1ambda commented:
4. `Apache Zeppelin` text need to be moved into the bottom a little bit while keeping the logo (icon) in the same position.

### Current

![image](https://user-images.githubusercontent.com/4968473/26961124-5fdd4526-4d17-11e7-8c7b-36ebb2423263.png)

1ambda commented:
5. (MINOR) we might add some shadows for texts in the section

I used this style, but we can test more different styles. 

```css
1px 1px #4a4a4a
```

### Current

![image](https://user-images.githubusercontent.com/4968473/26961214-17126596-4d18-11e7-9770-cbfac7683ae1.png)

### Fixed
![image](https://user-images.githubusercontent.com/4968473/26961202-00653f8a-4d18-11e7-866d-8e21786b60fb.png)

1ambda commented:
## 6. font-size, text-shadow for menus in the navbar including collapsed version.

### 6-1 normal version

Since the toplevel menu's font-size is `15px`, sub menus should have the equal or less font-size. currently, it's `16px`. 

- Let's set font-size `15px` for `.docs .dropdown-menu li span`

#### Current

![image](https://user-images.githubusercontent.com/4968473/26961495-1333e754-4d1a-11e7-816f-e48c1d5c0622.png)

#### Fixed

![image](https://user-images.githubusercontent.com/4968473/26961501-21c830c2-4d1a-11e7-8006-c68edf5dd50d.png)

### 6-2 collapsed version

- `text-shadow` in the sub menus can be decreased. too much IMO:  from `1px 2px #353131` to `1px 1px #353131`
- for `font-size`, we can apply the same policy with **6-1**
  * `15px` for `.navbar-inverse .navbar-nav > li > a ` (in all media queries for the collapsed version  just not for `@media only screen and (max-width: 996px) and (min-width: 768px)`
  * `15px` for `.docs .dropdown-menu b` (in all media queries  for the collapsed version )
  * `14px` for `.navbar-nav .open .dropdown-menu` (for all  for the collapsed version)
  * fix padding-bottom for `.docs .dropdown-menu li span` from `12px 10px 12px 20px` to `12px 10px 12px 15px`. Currently it's too close to children.
 
#### Current

![image](https://user-images.githubusercontent.com/4968473/26961635-3b6aaa7c-4d1b-11e7-924e-2f072309e5b2.png)

#### After

![image](https://user-images.githubusercontent.com/4968473/26961687-a732309a-4d1b-11e7-98b0-97150a99fc8e.png)

1ambda commented:
@soralee I left few 6 comments. Could you resolve one issue in a commit? It really helps to review.

soralee commented:
@1ambda Sure, Thank you for your helping always!

soralee commented:
@Leemoonsoo could you confirm this color? 
![chagne_button_color](https://user-images.githubusercontent.com/8110458/26965974-2d26e558-4d33-11e7-811d-29a1bda85d0d.gif)

1ambda commented:
@soralee IMO, using bright white as the color for not hovered button would be nicer. 
and let's find another color combinations.

what about others' opinion?

soralee commented:
@1ambda I've just updated as your comments except number 2 and 3. It might take time to resolve that. Let me inform if I finish to-do list. Thanks!

soralee commented:
I just improved No 2 and 3.
And, regarding No 3, there is not favicon.ico file. So, I added favicon.ico package files.
- not connected page
![image](https://user-images.githubusercontent.com/8110458/26977235-12485a30-4d62-11e7-84bd-98c12c7867ad.png)
- connected page
![image](https://user-images.githubusercontent.com/8110458/26977134-be4e2086-4d61-11e7-8610-d3530f3ee0f7.png)

1ambda commented:
@soralee 

- [ ] 1. Main text: resolved, but need others opinions.
- [x] 2. Nav doesn’t fold: fixed
- [x] 3. No favicon: fixed
- [x] 4. Narr magin-top: fixed
- [x] 5.Text shadow for main text: fixed
- [x] 6-1. Font for normal nav: fixed
- [ ] 6-2. Font for collapsed nav: top level font still has 18px font-size

1ambda commented:
Additionally, 

7. The color of main section is too bright. It doesn't go along nav color. Here are few options we can use.

- Add a color can be a bridge (e.g button or shadow for nav, ...)
- Use more deep blue and add border-shadow for nav to distinguish them easily.
- Or change nav color
- ... and more

![image](https://user-images.githubusercontent.com/4968473/27017839-8498c880-4f66-11e7-85bd-1c1ca3cf5f6d.png)

![image](https://user-images.githubusercontent.com/4968473/27017845-8b66419c-4f66-11e7-90d1-c45959d16165.png)

8. The color for buttons in main section

What about `#33a6cc` for background? The official website is already using the color for buttons below.

![image](https://user-images.githubusercontent.com/4968473/27017904-23e30b76-4f67-11e7-9f77-5a3998334f6c.png)

Here is a sample.

![image](https://user-images.githubusercontent.com/4968473/27017910-2ce9d916-4f67-11e7-8c8f-1662ed8e3e5a.png)

soralee commented:
@1ambda Thanks for opinions!
IMO, regarding of no.7, it would be better to match navigation bar with #2371.
Let's use light such as `box-shadow: 0px 3px 6px rgba(0, 0, 0, 0.4);` and revert to original color in main section.
What do you think?

And no.8 is a good idea. Let's change the buttons color!

soralee commented:
I couldn't change to original color in main section because, zeppelin log is disappeared :disappointed:. So changed one step more bright color than original main section color. 

![image](https://user-images.githubusercontent.com/8110458/27036170-f2632470-4fbe-11e7-8ad6-844914fda00a.png)

soralee commented:
Fixed 6.2 comments and, here is more improvement list
1. main section 
 - changed PR phrase to `font-size : 22px`, `line-height: 1.4`
 - added padding-top/bottom in main section
 - changed button font and weight
 - added button margin-side 
 - added button interval 
 - added interval between PR phrase in main section and buttons

2. revert to original what's new section

1ambda commented:
@soralee LGTM except for one thing,

h1 is hidden in mobile size browser.

![image](https://user-images.githubusercontent.com/4968473/27124501-3821ef96-512c-11e7-9007-1f6f38799ab5.png)

1ambda commented:
LGTM

soralee commented:
@1ambda Thank you for all your comment.

soralee commented:
Ping, Ready to review!

Leemoonsoo commented on _includes/themes/zeppelin/_navigation.html:
```diff
@@ -1,81 +1,98 @@
+    </div>
+    <nav class="navbar-collapse collapse" role="navigation">
+      <ul class="nav navbar-nav navbar-right">
+        <li><a href="/docs/0.8.0-SNAPSHOT/install/install.html">Quick Start</a></li>
```
I think this should link latest release version.

`/docs/latest/install/install.html`

soralee replied:
@Leemoonsoo Thanks for good point!
Let me update this and the following comment as well.

Leemoonsoo commented on _includes/themes/zeppelin/_navigation.html:
```diff
@@ -1,81 +1,98 @@
-        <a href="http://youtu.be/_PQbVH_aO5E" target="_zeppelinVideo" class="btn btn-primary btn-lg bigFingerButton" role="button">Tutorial Video</a>
-        <a href="./download.html" class="btn btn-primary btn-lg bigFingerButton" role="button">GET LATEST ZEPPELIN</a>
+      <div class="col-md-12 col-sm-12 col-xs-12 text-center newBtn" style="margin-top:8px;">
+        <a href="/docs/0.8.0-SNAPSHOT/install/install.html" class="btn btn-primary btn-lg" role="button"
```
This one, too. we need to point `latest` instead of `0.8.0-SNAPSHOT`

soralee replied:
updated it as well.

1ambda commented:
@soralee Since https://github.com/apache/zeppelin/pull/2371 was merged, we can fix the URL names in this PR too.

- quickstart button in nav: `${PROPER_PREFIX_FOR_0.8.0}/quickstart/install.html`

soralee commented:
@1ambda, @Leemoonsoo suggested the latest link so I already fix `0.7.2` version. 
@Leemoonsoo could you confirm this?

1ambda commented:
I think it's ok to use the prefix for 0.7.2. Then, we can deploy before 0.8.0

1ambda commented:
merge if no more discussion
